infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -581,5 +626,23 @@ public boolean isConnectionReceiver() {
     public void notifyLeadershipChange() {
         incomingMsgHandler.leadershipChanged();
     }
+
+
+    public void sendOutSessionToSinkSide(LogReplication.ReplicationSubscriber subscriber) {
+        Set<LogReplicationSession> sessionForSinkSide = this.createOutgoingSessionsBySubscriber(subscriber);
+        for (LogReplicationSession session : sessionForSinkSide) {
+            LogReplication.LogReplicationSinkSessionInitializationMsg msg =
+                    LogReplication.LogReplicationSinkSessionInitializationMsg.newBuilder().setSession(session).build();
+            CorfuMessage.RequestPayloadMsg payload =
+                    CorfuMessage.RequestPayloadMsg.newBuilder().setLrSinkSessionInitialization(msg).build();
+
+            updateRouterWithNewSessions();
+            createSourceFSMs();
+            logNewlyAddedSessionInfo();
+            topology.getRemoteClusterEndpoints().put(session.getSourceClusterId(), topology.getRemoteSourceClusters().get(session.getSourceClusterId()));
+            router.connect(router.getSessionToRemoteClusterDescriptor().get(session), session);
+            CompletableFuture<LogReplication.LogReplicationMetadataResponseMsg> cf = router.sendRequestAndGetCompletable(session, payload, router.getSessionToRemoteClusterDescriptor().get(session).getClusterId());
```

Comment:
Need to add the logic where we wait for an ACK here, and have a timeout. If the ACK doesn't arrive within the timeout, we resend the session msg.
Without this, we should not move to start the replication FSMs.

Reply:
+1